### PR TITLE
[ENH] Remove test exclusions

### PR DESCRIPTION
**Pull request by TonyBagnall**
new testing identified estimators that changed state in predict or transform. This was due to saving data in predict/transform to self.some_variable. Changed a couple to not store this data. If storing is required, it should be done in fit

### CODE REVIEW

File: aeon/segmentation/base.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def fit(self, X, y=None, axis=1):
         self.reset()
         if axis is None:  # If none given, assume it is correct.
             axis = self.axis
-        X = self._preprocess_series(X, axis, True)
```
Comment (MatthewMiddlehurst) [inline aeon/segmentation/base.py:119]: This should be fine for `fit`? Think thats the metadata param but may be wrong

**Review by MatthewMiddlehurst — COMMENTED** [review]
good chunk removed

File: aeon/segmentation/base.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/segmentation/base.py:119]: it is fine to store for fit, and  would prefer it, but iirc we dont store in collections? I'll check

Comment (TonyBagnall) [reply aeon/segmentation/base.py]: reverted to True

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**